src/pages/profile.tsx

```diff
@@ -0,0 +1,64 @@
+import { useSession } from 'next-auth/react';
+import { useEffect, useState } from 'react';
+import { SignOutButton } from '../components/SignOutButton';
+import { Image } from '@chakra-ui/react';
+
+export default function Profile() {
+  const { data: session, status } = useSession();
+  const [numAbsences, setNumAbsences] = useState<number | null>(null);
+  const [usedAbsences, setUsedAbsences] = useState<number | null>(null);
+
+  useEffect(() => {
+    const fetchUserInfoByEmail = async () => {
+      if (!session || !session.user || !session.user.email) return;
```

Comment:
I think you can check this ouside of the async block so it doesn't need to check every render cycle

Reply:
Do you mean within the `useEffect` but still outside of the `fetchUserDataByEmail`? The purpose of this check was to prevent the type safety from complaining about using `session.user.email`, so I think this check needs to stay here with how it's currently coded:

![image](https://github.com/user-attachments/assets/8aaae46c-4e11-4d2b-a333-3557abe30271)